src/tree.rs

```diff
@@ -444,6 +444,49 @@ impl<T> Tree<T> {
         Result::Ok(())
     }
 
+    pub fn replace(&mut self, node_id: &NodeId, mut new_node: Node<T>) -> Result<Node<T>, NodeIdError>
+    {
+        use std::mem;
+
+        let (is_valid, error) = self.is_valid_node_id(node_id);
+        if !is_valid {
+            return Result::Err(error.expect("Tree::replace: Missing an error value on finding an invalid NodeId."));
+        }
+
+        // lets cheat a bit and return new_node ;)
+        // TODO, while this is elegant, this worksaround NodeBuilder's Parameters
+        // Should we drain the old children array instead?
+        mem::swap(self.get_mut_unsafe(node_id).data_mut(), new_node.data_mut());
+
+        Result::Ok(new_node)
+    }
+
+    pub fn swap(&mut self, first_id: &NodeId, second_id: &NodeId) -> Result<(), NodeIdError> {
```

Comment:
I'm having second thoughts about the name on this one.

To me (and maybe this is because of our discussion earlier about `mem::swap`) `swap` sounds like it would just swap the data in one `Node` with the data in another one (both of which are currently in the `Tree`).  While I think that would be a neat feature to have, that's not the intent with this function (correct?).

If we're moving the sub-tree around with the `Node` I almost want to call it `swap_with_children` or `swap_sub_trees` or something to that effect.  I'm not necessarily a fan of either of those, but I think they're more clear.  Thoughts?